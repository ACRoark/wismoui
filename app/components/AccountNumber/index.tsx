import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
