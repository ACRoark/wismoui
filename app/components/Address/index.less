@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
