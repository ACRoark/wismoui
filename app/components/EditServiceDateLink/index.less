@import (reference) '../../styles/themes/index.less';

@edit-cls-prefix: ~'@{dte-prefix}-edit-service-date-link';

.@{edit-cls-prefix} {
  > button {
    font-size: @dte-wismo-font-size-0;
    font-weight: @dte-wismo-font-weight-semi-bold;
    line-height: 1.3rem;
    margin-left: 22px;
    padding: 0px;
  }

  &-button:hover {
    border-color: @dte-blue;
  }

  &-cancel-button {
    margin-right: 20px;
  }

  &-modal > .@{dte-prefix}-modal-body > .@{dte-prefix}-modal-content {
    min-height: 0px;
    padding-bottom: 20px;
  }
}

.@{dte-prefix}-edit-service-date-link-button {
  margin-top: 20px;

  @media @mobile {
    margin-top: 0;
  }
}

.content-edit-date {
  text-align: left;
}

.secondary-button-edit-date {
  margin-right: 20px;

  @media @mobile {
    margin: 20px 0px;
  }
}

.modal-buttons {
  @media @mobile {
    display: flex;
    flex-direction: column;
  }
}
