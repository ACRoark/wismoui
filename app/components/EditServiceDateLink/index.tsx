import moment from 'moment';
import React from 'react';

import { userIsAuthenticated } from 'security/utils';

import './index.less';

import EditLink from './EditLink';

interface IEditServiceDateLinkProps {
  wantDate: string;
}

const EditServiceDateLink: React.FC<IEditServiceDateLinkProps> = (
  props: IEditServiceDateLinkProps,
): React.ReactElement => {
  const { wantDate } = props;

  const today = moment();
  const scheduledDate = moment(wantDate);

  const showLink = userIsAuthenticated() && scheduledDate.isAfter(today);

  return <>{showLink && <EditLink />}</>;
};

export default EditServiceDateLink;
