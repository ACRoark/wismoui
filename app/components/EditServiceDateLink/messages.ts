import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.edit-service-date-link';

export default defineMessages({
  cancelButtonText: {
    id: `${scope}.cancel-button-text`,
  },
  continueButtonText: {
    id: `${scope}.continue-button-text`,
  },
  linkText: {
    id: `${scope}.link-text`,
  },
  promptMessage: {
    id: `${scope}.prompt-message`,
  },
  promptTitle: {
    id: `${scope}.prompt-title`,
  },
});
