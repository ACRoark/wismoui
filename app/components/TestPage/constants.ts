import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
