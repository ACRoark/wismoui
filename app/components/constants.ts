import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
