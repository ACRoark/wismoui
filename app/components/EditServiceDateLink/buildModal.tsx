import React from 'react';
import { FormattedMessage } from 'react-intl';

import Button from 'components/Button';
import IModalConfig from 'types/IModalConfig';

import './index.less';
import messages from './messages';

const buildModal = (onCancel: () => void, onContinue: () => void): IModalConfig => ({
  content: (
    <div className="content-edit-date">
      <FormattedMessage {...messages.promptMessage} />
    </div>
  ),
  footer: (
    <div className="modal-buttons">
      <Button
        className="secondary-button-edit-date dte-wismo-edit-service-date-link-button"
        type="secondary"
        onClick={onCancel}
      >
        <FormattedMessage {...messages.cancelButtonText} />
      </Button>
      <Button className="dte-wismo-edit-service-date-link-button" type="primary" onClick={onContinue}>
        <FormattedMessage {...messages.continueButtonText} />
      </Button>
    </div>
  ),
  title: <FormattedMessage {...messages.promptTitle} />,
});

export default buildModal;
