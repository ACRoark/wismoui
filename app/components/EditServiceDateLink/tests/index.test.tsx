import React from 'react';
import { Provider } from 'react-redux';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import useModal from 'hooks/useModal';
import { userIsAuthenticated } from 'security/utils';
import { createMockStore, createSnapshotWithIntl, renderWithIntl, today, tomorrow, yesterday } from 'testing/utils';

import EditServiceDateLink from '..';

jest.mock('hooks/useConfig');
jest.mock('hooks/useModal');
jest.mock('security/utils');

const mockAuth = userIsAuthenticated as jest.MockedFunction<typeof userIsAuthenticated>;
const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;
const mockModal = useModal as jest.MockedFunction<typeof useModal>;

const itIsToday = today().format('YYYY-MM-DD');
const itIsTomorrow = tomorrow().format('YYYY-MM-DD');
const itIsYesterday = yesterday().format('YYYY-MM-DD');

const store = createMockStore();

const expectComponentToMatchSnapshot = (language: string, date: string, isAuthenticated: boolean): void => {
  mockAuth.mockReturnValue(isAuthenticated);
  mockConfig.mockReturnValue({ ...testConfig });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const tree = createSnapshotWithIntl(<Provider store={store}><EditServiceDateLink wantDate={date}/></Provider>, language);

  expect(tree).toMatchSnapshot();
};

const shouldNotLogErrors = (language: string, date: string, isAuthenticated: boolean): void => {
  mockAuth.mockReturnValue(isAuthenticated);
  mockConfig.mockReturnValue({ ...testConfig });
  mockModal.mockReturnValue({
    hideModal: jest.fn(),
    showModal: jest.fn(),
  });

  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<Provider store={store}><EditServiceDateLink wantDate={date}/></Provider>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('<EditServiceDateLink />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when the scheduled service date is in the future and the user is authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsTomorrow, true));

    it(`should not log errors in console when the scheduled service date is in the future and the user is not authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsTomorrow, false));

    it(`should not log errors in console when the scheduled service date is in the past and the user is authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsYesterday, true));

    it(`should not log errors in console when the scheduled service date is in the past and the user is not authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsYesterday, false));

    it(`should not log errors in console when the scheduled service date is today and the user is authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsToday, true));

    it(`should not log errors in console when the scheduled service date is today and the user is not authenticated (${language})`, (): void =>
      shouldNotLogErrors(language, itIsToday, false));

    it(`should not render the link when the scheduled service date is in the future and the user is not authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsTomorrow, false));

    it(`should not render the link when the scheduled service date is in the past and the user is authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsYesterday, true));

    it(`should not render the link when the scheduled service date is in the past and the user is not authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsYesterday, false));

    it(`should not render the link when the scheduled service date is today and the user is authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsToday, true));

    it(`should not render the link when the scheduled service date is today and the user is not authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsToday, false));

    it(`should render the link when the scheduled service date is in the future and the user is authenticated (${language})`, (): void =>
      expectComponentToMatchSnapshot(language, itIsTomorrow, true));
  });
});
