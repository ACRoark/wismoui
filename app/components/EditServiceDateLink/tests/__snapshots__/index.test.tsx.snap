// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the future and the user is not authenticated (en) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the future and the user is not authenticated (es) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the past and the user is authenticated (en) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the past and the user is authenticated (es) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the past and the user is not authenticated (en) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is in the past and the user is not authenticated (es) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is today and the user is authenticated (en) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is today and the user is authenticated (es) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is today and the user is not authenticated (en) 1`] = `null`;

exports[`<EditServiceDateLink /> should not render the link when the scheduled service date is today and the user is not authenticated (es) 1`] = `null`;

exports[`<EditServiceDateLink /> should render the link when the scheduled service date is in the future and the user is authenticated (en) 1`] = `
<div
  className="dte-wismo-edit-service-date-link"
>
  <button
    className="dte-wismo-button dte-wismo-button-link"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;

exports[`<EditServiceDateLink /> should render the link when the scheduled service date is in the future and the user is authenticated (es) 1`] = `
<div
  className="dte-wismo-edit-service-date-link"
>
  <button
    className="dte-wismo-button dte-wismo-button-link"
    data-track="mimo"
    data-track-action="click"
    data-track-detail="order tracker"
    onClick={[Function]}
  >
    Editar
  </button>
</div>
`;
